proposals/21-11-transformer-annotations.md

```diff
@@ -0,0 +1,253 @@
+    type: custom
+  - kind: NamespaceTransformer
+    path: examples/multibases/base/kustomization.yaml
+    repo: https://github.com/kubernetes-sigs/kustomize
+    ref: v1.0.6 
+    type: builtin
+```
+
+### Kustomize edit
+
+We will want to provide convenient commands for users to edit the `buildMetadata` field in their kustomization.
+We should support the following commands:
+
+`kustomize edit add buildMetadata originAnnotations`
+
+`kustomize edit add buildMetadata transformerAnnotations`
+
+`kustomize edit add buildMetadata *`
+
+`kustomize edit remove buildMetadata originAnnotaitons`
+
+`kustomize edit remove buildMetadata transformerAnnotations`
+
+`kustomize edit remove buildMetadata *`
+
+The wildcard match `*` should add/remove all possible options for `buildMetadata`.
+ 
+
+### User Stories
+
+#### Story 1
+
+I am running kustomize build on some configuration that has already been created. This configuration has a large number
+of resources and patches through multiple overlays in different directories. After running `kustomize build`, I get a 
+stream of 100+ resources, but I notice that a few of them have a typo, likely originating from the same base or patch 
+file. However, because my directory structure is large and complex, it is difficult to narrow down which base these
+resources originated from and which patches have been applied to it. In the top-level kustomization file, I can add a
+new `buildMetadata` field:
+
+```yaml
+buildMetadata: [originAnnotations, transformerAnnotations]
+```
+
+The output of `kustomize build` will contain patched resources with annotations similar to the following:
+```
+annotations:
+  config.kubernetes.io/origin: |
+    path: deployment.yaml
+  config.kubernetes.io/transformations:
+  - kind: PatchTransformer
+    path: ../base/kustomization.yaml
+    type: builtin
+  - kind: PatchStrategicMergeTransformer
+    path: ../dev/patch.yaml
+    type: builtin
+  - kind: PatchesJson6902Transformer
+    path: patch.yaml
+    type: builtin
+```
+
+From these annotations, I can narrow down the set of files that contributed to the resource's final output and
+debug it accordingly. 
+
+### Risks and Mitigations
+N/A
+
+### Dependencies
+N/A
+
+### Scalability
+N/A
+
+## Drawbacks
+N/A
+
+## Alternatives
+We considered putting this data in the comments of the rendered resources, but this would be a large project because 
+kustomize doesn't currently support comments. 
+
+We also considered making this a flag to `kustomize build` rather than additional field, but we would like to align 
+with the kustomize way of avoiding build-time side effects and have everything declared explicitly in the kustomization.
+
+## Rollout Plan
+This is a fairly lightweight, minor and optional feature that can be implemented and
+released without needing a rollout plan, much like the `originAnnotations` option that
+we are extending. With sufficient testing, there is no need to hide this feature behind 
+an alpha flag. 
```

Comment:
I agree we don't need a flag for this, but wdyt about starting the annotation in alpha?

Reply:
We can have it start in alpha. How do we mark things as alpha in kustomize?

Reply:
I was just thinking `alpha.config.kubernetes.io/transformations` 🤷‍♀️